File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    void solve(vector<int> nums, vector<int> res, vector<vector<int>> & ans, int target, int i){
        if(target == 0){
            ans.push_back(res);
            return;
        }
        if(i == nums.size()) return;
        if(target < 0) return;

        solve(nums, res, ans, target, i+1);
        res.push_back(nums[i]);
        solve(nums,res, ans, target - nums[i], i);

        return;
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> ans;
        vector<int> res;
        solve(nums, res, ans, target, 0);
        return ans;
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, h = nums.length-1;
        while(l <= h){
            int mid = (l + h)/2;
            if(nums[mid] == target){
                return mid;
            }
            else if(nums[mid] > target){
                h--;
            }
            else{
                l++;
            }
        }
        return -1;
    }
}
